pkg/jobrunaggregator/jobrunhistoricaldataanalyzer/analyzer.go

```diff
@@ -59,20 +61,34 @@ func (o *JobRunHistoricalDataAnalyzerOptions) Run(ctx context.Context) error {
 		return fmt.Errorf("new historical data is empty, can not compare")
 	}
 
-	// We check to make sure the current data set doesn't contain the previous release version
-	// If that's the case, then we are in a new release event (i.e. a new branch has been cut)
-	// We then write to a `require_review` file to record why a review would be required.
+	// We convert our query data to maps to make it easier to handle
+	newDataMap := convertToMap(newHistoricalData)
+	currentDataMap := convertToMap(currentHistoricalData)
+
+	// We check to see if we need to transition to use new release data. We decide based on:
+	// 1. The current data set contains the previous release version
+	// 2. We have enough data count from the new release
+	// If that's the case, we will transition to use data set from the new release.
+	// We also write to a `require_review` file to record why a review would be required.
 	newReleaseUpdate := currentDataContainsPreviousRelease(previousRelease, currentHistoricalData)
 	if newReleaseUpdate {
-		if err := requireReviewFile("The current data contains previous release version\n"); err != nil {
-			return err
+		newReleaseDataCount := 0
+		for _, data := range newDataMap {
+			if data.GetJobData().Release == currentRelease {
+				newReleaseDataCount++
+			}
+		}
+		if newReleaseDataCount < len(currentDataMap)*releaseTransitionDataCountThreshold/100 {
```

Comment:
The change looks good Ken, our minimum 100 entries looks to happen in the query, so it's already been applied by the time we get to this code, and your logic here should hold, if we didn't have at least 60% of entries with 100+ runs, we stick to old release.

Reply:
I would add a very clear output message here for printing, indicate the old and new counts with their release version, and that we require at least X percent, and are sticking with the previous version.

Is there any way we could also get that message into the PR message so the reviewer gets a clear indication as well? They should probably notice anyhow but still, it would be nice if it's not too difficult.